collector/collector.go

```diff
@@ -244,3 +244,11 @@ func pushMetric(ch chan<- prometheus.Metric, fieldDesc *prometheus.Desc, name st
 
 	ch <- prometheus.MustNewConstMetric(fieldDesc, valueType, fVal, labelValues...)
 }
+
+func uint64PtrToFloat(u *uint64) float64 {
+	if u == nil {
+		return float64(0)
+	}
```

Comment:
We definitely don't want to do this. The reason the procfs library returns nil is so that we know not to expose metrics that don't exist.

This will generate metrics with zero values when they don't exist on a system.

Reply:
Addressed in rebased commits :+1: